pkg/controller/components/apiproxy.go

```diff
@@ -0,0 +1,166 @@
+package components
+
+import (
+	"fmt"
+
+	networkingv1 "k8s.io/api/networking/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/dominodatalab/distributed-compute-operator/pkg/cluster/metadata"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/actions"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
+)
+
+const (
+	defaultAPIProxyPort        = 8899
+	targetServicePort          = 8899
+	apiProxyPortName           = "api-proxy"
+	component                  = "api-proxy"
+	executionIDLabel           = "dominodatalab.com/execution-id"
+	projectIDLabel             = "dominodatalab.com/project-id"
+	datasourceProxyClientLabel = "datasource-proxy-client"
+)
+
+func executionID(obj *client.Object) string {
+	if executionID, ok := (*obj).GetLabels()[executionIDLabel]; ok {
+		return executionID
+	}
+	return "unknown-execution-id" // primarily, for integration testing
+}
+
+func runPodSelector(obj *client.Object) map[string]string {
+	runPodName := fmt.Sprintf("run-%s", executionID(obj))
+	return map[string]string{
+		metadata.ApplicationInstanceLabelKey: runPodName,
+	}
+}
+
+func newResourceMeta(obj *client.Object, componentMeta *metadata.Provider) metav1.ObjectMeta {
+	instanceName := fmt.Sprintf("%s-%s", component, executionID(obj))
```

Comment:
Can you use `meta.InstanceName(s.dc, s.comp)`?

Reply:
Yes, makes sense.